### PR TITLE
Support `@IdClass` and multiple `@Id`

### PR DESCRIPTION
**Pull request by dstepanov**

### CODE REVIEW

File: data-model/src/main/java/io/micronaut/data/model/PersistentEntity.java [hunk 1]
```diff
@@ -88,6 +87,20 @@ default boolean hasIdentity() {
...
     @Nullable PersistentProperty getIdentity();
 
+    /**
+     * Returns all identity properties.
+     *
+     * @return The identity properties
+     */
+    @NonNull
+    default List<PersistentProperty> getIdentityProperties() {
+        if (getIdentity() != null) {
```
Comment (radovanradic) [inline data-model/src/main/java/io/micronaut/data/model/PersistentEntity.java:97]: Should look at `getCompositeIdentity()` ?

File: data-model/src/main/java/io/micronaut/data/model/runtime/RuntimePersistentEntity.java [hunk 2]
```diff
@@ -306,6 +306,15 @@ public RuntimePersistentProperty<T> getIdentity() {
         return identity.length == 1 ? identity[0] : null;
     }
 
+    @Override
+    public List<PersistentProperty> getIdentityProperties() {
+        return List.of(identity);
+    }
+
+    public List<RuntimePersistentProperty<T>> getRuntimeIdentityProperties() {
```
Comment (radovanradic) [inline data-model/src/main/java/io/micronaut/data/model/runtime/RuntimePersistentEntity.java:318]: Is this duplicate/not needed?

File: data-jdbc/src/test/java/io/micronaut/data/jdbc/postgres/PostgresEntityWithIdClassRepository.java [hunk 3]
```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright 2017-2020 original authors
```
Comment (radovanradic) [inline data-jdbc/src/test/java/io/micronaut/data/jdbc/postgres/PostgresEntityWithIdClassRepository.java:2]: Not really important, but maybe we should just remove copyright from test classes? If not then update from 2020 to the current year.

File: data-model/src/main/java/io/micronaut/data/model/runtime/RuntimePersistentEntity.java (hunk 2 above)
Comment (dstepanov) [reply data-model/src/main/java/io/micronaut/data/model/runtime/RuntimePersistentEntity.java:318]: It's returning runtime properties, unfortunately the original method cannot have a different type

File: data-jdbc/src/test/java/io/micronaut/data/jdbc/postgres/PostgresEntityWithIdClassRepository.java (hunk 3 above)
Comment (dstepanov) [reply data-jdbc/src/test/java/io/micronaut/data/jdbc/postgres/PostgresEntityWithIdClassRepository.java:2]: Probably, I just copied :/

File: data-model/src/main/java/io/micronaut/data/model/PersistentEntity.java (hunk 1 above)
Comment (dstepanov) [reply data-model/src/main/java/io/micronaut/data/model/PersistentEntity.java:97]: I will try to add it an see if it's going to work

Comment (radovanradic) [reply data-model/src/main/java/io/micronaut/data/model/PersistentEntity.java:97]: I guess check if getCompositeIdentity is not empty then return it, else list with getIdentity or else empty list
To have same logic as implementing classes I guess.

**Review by radovanradic — APPROVED** [review]

**Pull request merged**